pkg/env/metrics.go

```diff
@@ -1,31 +1,90 @@
+	MetricsPort = RegisterSetting("ROX_METRICS_PORT", WithDefault(defaultHTTPPort))
+	// SecureMetricsPort has the :port or host:port string for listening for metrics/debug server.
+	SecureMetricsPort = RegisterSetting("ROX_SECURE_METRICS_PORT", WithDefault(defaultHTTPSPort))
+	// SecureMetricsCertDir has the server's TLS certificate files.
+	SecureMetricsCertDir = RegisterSetting("ROX_SECURE_METRICS_CERT_DIR", WithDefault("/run/secrets/stackrox.io/monitoring-tls"))
+	// SecureMetricsClientCANamespace has the namespace that contains the client CA.
+	SecureMetricsClientCANamespace = RegisterSetting("ROX_SECURE_METRICS_CLIENT_CA_NS", WithDefault("kube-system"))
+	// SecureMetricsClientCAConfigMap has the config map that contains the client CA.
+	SecureMetricsClientCAConfigMap = RegisterSetting("ROX_SECURE_METRICS_CLIENT_CA_CFG", WithDefault("extension-apiserver-authentication"))
 )
 
-// ValidateMetricsSetting returns an error if the environment variable is invalid.
-func ValidateMetricsSetting() error {
-	val := MetricsSetting.Setting()
-	if val == "disabled" {
-		return nil
-	}
+func validatePort(setting Setting) error {
+	val := setting.Setting()
 	addr, err := net.ResolveTCPAddr("tcp", val)
 	if err != nil {
 		return err
 	}
-	log.Debugf("%s=%s, resolved to %+v", MetricsSetting.EnvVar(), val, addr)
+	log.Debugf("%s=%s, resolved to %+v", setting.EnvVar(), val, addr)
 	return nil
 }
 
-// MetricsEnabled returns true if the metrics/debug http server should be started
+func validateTLS() error {
+	certFile := filepath.Join(SecureMetricsCertDir.Setting(), TLSCertFileName)
+	if ok, err := fileutils.Exists(certFile); !ok {
+		if err != nil {
+			log.Warnf("failed to validate secure metrics certificate file %q", certFile)
+		}
+		return errors.Errorf("secure metrics certificate file %q not found", certFile)
+	}
+
+	keyFile := filepath.Join(SecureMetricsCertDir.Setting(), TLSKeyFileName)
+	if ok, err := fileutils.Exists(keyFile); !ok {
+		if err != nil {
+			log.Warnf("failed to validate secure metrics key file %q", keyFile)
+		}
+		return errors.Errorf("secure metrics key file %q not found", keyFile)
+	}
+	return nil
+}
+
+// ValidateMetricsSetting returns an error if the environment variable is invalid.
+func ValidateMetricsSetting() error {
+	if !MetricsEnabled() {
+		return nil
+	}
+	return validatePort(MetricsPort)
+}
+
+// MetricsEnabled returns true if the metrics/debug http server should be started.
 func MetricsEnabled() bool {
-	return MetricsSetting.Setting() != "disabled"
+	return MetricsPort.Setting() != "disabled"
+}
+
+// ValidateSecureMetricsSetting returns an error if the environment variable is invalid.
+func ValidateSecureMetricsSetting() error {
+	if !SecureMetricsEnabled() {
+		return nil
+	}
+	if err := validateTLS(); err != nil {
+		return err
+	}
+	return validatePort(SecureMetricsPort)
+}
+
+// SecureMetricsEnabled returns true if the metrics/debug https server should be started.
+func SecureMetricsEnabled() bool {
```

Comment:
It's a bit unfortunate that we have this conditionally being a value set to `ROX_SECURE_METRICS_PORT`, other environment variables specifically add another boolean environment variable to it. I get why that may be the case for the metrics port, but for the newly introduce one, should it be also "disabled" or rather a boolean flag additionally?